### ISSUE TITLE
Would you like to modify the boost::container::flat_set<size_t> to std::vector?

### ISSUE THREAD

**Issue by zhangfq-chemistry** [ilastik/marching_cubes#36]
I do not install boost under windows. Would you please to slightly change the Adjacency from boost to stl? Thanks a lot!

**Comment by stuarteberg** [comment]
>Would you like to modify the `boost::container::flat_set<size_t>` to `std::vector`?

No, `flat_set` is not equivalent to `vector`, so that wouldn't work in this case.  Perhaps `std::set` would work, but that would likely be much slower than `flat_set` for this use-case.

>I do not install boost under windows.

In this case, the installation should be trivial.  `boost::container` is a [header-only library][1], so there is no need to compile anything.  Just put the library's source code in a directory your compiler will search when looking for header files.

[1]: https://www.boost.org/doc/libs/1_76_0/doc/html/container.html#container.intro.introduction_building_container

**Comment by DerThorsten** [comment]
@stuarteberg years ago I wrote  flat set on my own for OpenGM which was conserved / polished by @bjoern-andres

https://github.com/bjoern-andres/random-access-set/blob/master/include/andres/random-access-set.hxx

We might want to use that
